lib/backend/info/addressDetails.ts

```diff
@@ -0,0 +1,53 @@
+
+   
+class AddressDetails{
+    location: string;
+    apartment: string;
+    addressLine: string;
+    state: string;
+
+    constructor(location: string, apartment: string, addressLine: string, state: string){
+        this.location = location;
+        this.apartment = apartment;
+        this.addressLine = addressLine;
+        this.state = state;
+    }
+}
+
+class AddressDetailsBuilder{
+    location: string;
+    apartment: string;
+    addressLine: string;
+    state: string;
+
+    constructor(){
+        return this;
```

Comment:
for this, I think apartment can be set as default to "", but others are required.